TerriaJS/terriajs
# Zoom status indicators

### What this PR does

- [x] Adds a toast and zoom button spinner when map is zooming similar to the attached design.
- [x] Modifies the `zoomTo()` call to return a promise that resolves when the zoom animation is complete.
- [x] Implement slide in/out animation for the toast
- [x] Fixed toast positioning on mobile - shown 40px from bottom left.

![image](https://user-images.githubusercontent.com/1430/114814731-23065580-9df8-11eb-98ed-43d805d7d64e.png)

### Testing

* Visit http://ci.terria.io/zoom-status-indicators/#share=s-hQ3zR006Fur8ObS3fZh1fiveM4V
* Click ideal zoom for the first catalog item. 
* The zoom button should show a spinner and a toast should appear at bottom center of the screen
* The zoom button will be disabled while the zoom is in progress
* Click ideal zoom for the second catalog item
* You should see the same results


### Checklist
